console/console-init/ui/src/modules/iot-device/utils/TableFormatter.tsx

```diff
@@ -24,26 +25,14 @@ const renderDeviceType = (type: string) => {
 };
 
 export const getTableCells = (row: IDevice) => {
-  const {
-    enabled,
-    via,
-    credentials,
-    deviceId,
-    updated,
-    created,
-    lastSeen
-  } = row;
+  const { enabled, via, viaGroups, deviceId, updated, created, lastSeen } = row;
 
-  const viaGateway = via && via?.length > 0;
+  let deviceType: DeviceConnectionType;
 
-  let deviceType: string;
-
-  if (!via?.length === !credentials?.length) {
-    deviceType = "N/A";
-  } else if (viaGateway) {
-    deviceType = "Via gateways";
+  if (via?.length || viaGroups?.length) {
```

Comment:
@jbtrystram since we can't get the credentials in the devices query, there is no way to check if a device has conflicting type, we will just be able to check the lengths of via & viaGroups and then determine the deviceType, is that okay.